forge-game/src/main/java/forge/game/ability/effects/SpecializeEffect.java

```diff
@@ -0,0 +1,51 @@
+package forge.game.ability.effects;
+
+import com.google.common.collect.Lists;
+import forge.card.MagicColor;
+import forge.game.ability.AbilityUtils;
+import forge.game.ability.SpellAbilityEffect;
+import forge.game.card.Card;
+import forge.game.player.Player;
+import forge.game.spellability.SpellAbility;
+import forge.util.Localizer;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+
+public class SpecializeEffect extends SpellAbilityEffect {
+    @Override
+    public void resolve(SpellAbility sa) {
+        final Player player = sa.getActivatingPlayer();
+        final Card discarded = sa.getPaidList("Discarded").get(0);
+
+        List<String> colorChoices = Lists.newArrayList();
+        for (final String col : MagicColor.Constant.ONLY_COLORS) {
+            if (discarded.isOfColor(col)) {
+                colorChoices.add(col);
+            }
+        }
+        if (discarded.isLand()) {
+            if (discarded.getType().hasSubtype("Plains") && !colorChoices.contains("white")) {
+                colorChoices.add("white");
+            }
+            if (discarded.getType().hasSubtype("Island") && !colorChoices.contains("blue")) {
+                colorChoices.add("blue");
+            }
+            if (discarded.getType().hasSubtype("Swamp") && !colorChoices.contains("black")) {
+                colorChoices.add("black");
+            }
+            if (discarded.getType().hasSubtype("Mountain") && !colorChoices.contains("red")) {
+                colorChoices.add("red");
+            }
+            if (discarded.getType().hasSubtype("Forest") && !colorChoices.contains("green")) {
+                colorChoices.add("green");
+            }
+        }
+
+        String chosenColor = player.getController().chooseColors(
+                Localizer.getInstance().getMessage("lblChooseAColor"), sa, 1, 1, colorChoices).get(0);
+        String code = chosenColor.equals("blue") ? "U" : StringUtils.capitalize(chosenColor.substring(0, 1));
```

Comment:
i don't like the use of getting first letter, can't we use some ManaColor function to get it?

also what happens when you discard a colorless card? nothing?

Reply:
Yeah, I figured there was a better way.

Looks like they avoid the colorless thing this way, so will need to do some reworks for sure:
![image](https://user-images.githubusercontent.com/103371817/184497693-648b99b5-280f-41cc-8287-51a7a1abcad6.png)

Reply:
Choosing the color is part of the cost 🤢 ??

Reply:
This is the wild west for rules anyway, so I think I'll just make that discard part of specialize effect after color choice. Color choices can just be filtered by colors in hand.

I don't know what would happen if your entire hand was colorless/no basic land types... probably you just can't specialize?

Reply:
Eh that's easy, you can choose a color but you can't pay the discard cost, e.g. you can't Specialize

Reply:
Resolving this, as I don't think we'll need a new API